framework/audio/engine/internal/audioengine.cpp

```diff
@@ -117,39 +117,100 @@ IAudioEngine::handle AudioEngine::play(std::shared_ptr<IAudioSource> s, float vo
     }
 
     handle h = m_sl->engine.play(*sa, volume, pan, paused);
+
+    pushMeta(h);
+
+    if (paused) {
+        onPause(h);
+    } else {
+        onPlay(h);
+    }
+
     return h;
 }
 
 void AudioEngine::seek(handle h, time sec)
 {
     m_sl->engine.seek(h, sec);
+    onSeek(h);
+}
+
+void AudioEngine::setPause(handle h, bool paused)
+{
+    m_sl->engine.setPause(h, paused);
+    if (paused) {
+        onPause(h);
+    } else {
+        onPlay(h);
+    }
 }
 
 void AudioEngine::stop(handle h)
 {
     m_sl->engine.stop(h);
+    //! NOTE No need to call `onStop`, it will be called when the instance is destroyed
 }
 
-void AudioEngine::setPause(handle h, bool paused)
+IAudioEngine::Status AudioEngine::status(handle h) const
 {
-    m_sl->engine.setPause(h, paused);
+    return meta(h).status;
 }
 
-void AudioEngine::stopAll()
+mu::async::Channel<IAudioEngine::Status> AudioEngine::statusChanged(handle h) const
 {
-    m_sl->engine.stopAll();
+    return meta(h).statusChanged;
 }
 
-IAudioEngine::time AudioEngine::position(handle h) const
+void AudioEngine::onPlay(handle h)
 {
-    return m_sl->engine.getStreamPosition(h);
+    HandleMeta& m = meta(h);
+    IF_ASSERT_FAILED(m.isValid()) {
+        return;
+    }
+    m.status = Status::Playing;
+    m.statusChanged.send(m.status);
+}
+
+void AudioEngine::onSeek(handle h)
+{
+    HandleMeta& m = meta(h);
+    IF_ASSERT_FAILED(m.isValid()) {
+        return;
+    }
+    //! NOTE We do not change the status, we only notify that it was seek
+    m.statusChanged.send(Status::Seeked);
```

Comment:
I'm trying to figure out, who needs this kind of notification?

Reply:
Not for anyone at the moment.
Made for the integrity of the picture.
Do you consider it unnecessary?

Reply:
Yeah, I think it's redundant and may confuse people

Reply:
ok, removed